### PR TITLE
Add unstable `initializer` function

### PR DESCRIPTION
**Pull request by JakubBachurskiQC**
This is a proposal for an opset-independent `initializer` function in `_future`, built using ONNX initialisers.

```py
def initializer(value: npt.ArrayLike, dtype: npt.DTypeLike = None) -> Var:
    ...
```

It follows the same signature as the new version of the extra `ai.onnx::const` (#45), except that it does not depend on `ai.onnx` - and as such does not fix an opset version.

Initialisers have been implemented in Spox for a while but have seen limited usage. Starting to use them is a step towards having a more complete public API & other interesting uses (like #25).

---

Even though this proposal now places `initializer` in `_future`, it remains used in `inline`, as I think it's an appropriate place (it's the first occurrence where we had to construct an opset-independent constant).

Note the branch name is now outdated, as the new function is not stable.

### CODE REVIEW

**Comment by JakubBachurskiQC** [comment]
To clarify, on a technical level initialisers seem to have always been about having a global list of constants as you note. Then the ability to make them behave like default model inputs was added, but it remains unsupported by ORT (due to likely a bug).

I feel similarly that there's something off about applying them in terms of output quality. As I mentioned to you, it seems like `onnx.parser`/`onnx.printer` don't actually support initialisers. 

I'm not fully sure whether I'm happy with it becoming entirely public, maybe it should indeed remain named `initializer` and land in `spox._future` until further notice. `op.const` should serve most needs.

**Comment by JakubBachurskiQC** [comment]
@cbourjau This should now be ready for review to be added to `spox._future`. It is applied in `inline` when constant values have to get constructed, replacing the previous mechanism with adding versioned `Constant` operators (in the specific use case of default argument values).

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,21 @@
...
+
+- The opset ``ai.onnx@18`` is now shipped with Spox (version 18 of the default domain, as introduced in ONNX 1.13). To avoid code duplication, unchanged implementations are 'inherited' from the previous version.
+
+**Bug fixes**
+
+- The operator constructor ``cum_sum`` now also has an alias ``cumsum``, to mirror ``numpy``. This alias should be preferred.
+
+**New unstable features**
+
+- ``spox._future.initializer(ArrayLike, DTypeLike = None) -> Var`` is a new function for creating of variables from a constant value. The created value is constructed with ``numpy.array``. This function is opset-independent and is implemented using initializers.
```
Comment (cbourjau) [inline CHANGELOG.rst:23]:
```suggestion
- ``spox._future.initializer(ArrayLike, DTypeLike = None) -> Var`` is a new function for creating variables from a constant value. The created value is constructed with ``numpy.array``. This function is opset-independent and is implemented using initializers.
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,6 +7,21 @@
 Change log
 ==========
 
+0.7.0 (2023-04-xx)
+------------------
+
+**New features**
```
Comment (cbourjau) [inline CHANGELOG.rst:13]: Was it deliberate to put all `0.7` changelog entries here?

Comment (JakubBachurskiQC) [reply CHANGELOG.rst:13]: Yes, I forgot to do it on the PR for v18 I had just merged.

**Pull request merged**